### ISSUE TITLE
Entities don't come back after remote docker comes back online

### ISSUE THREAD

**Issue by defaultsecurity** [ualex73/monitor_docker#165]
Thank you for your hard work. The integration is amazing and very useful.

I'm running docker on a remote computer on local LAN.
- Connection to Home Assistant is good.
- Entities are great.
- I can switch containers on and off without any problems.

**The problem**
- When I turn the remote computer off, ALL entities disappear. (except for update.monitor_docker_update)
- After the remote computer comes back online, the entities stay non-existent and don't come back.
- I have to restart Home Assistant to get the entities back.

**Comment by jerome-labidurie** [comment]
looks like a duplicate of #103

**Comment by ualex73** [comment]
Can you try the 1.20b0 (pre-release)? It should resolve this.

**Comment by ualex73** [comment]
BTW, entities still disappear when the host is rebooted ... but they should come back after the host is online.

**Comment by jerome-labidurie** [comment]
hello,
Tried the 1.20b1 with a remote host on a fresh HA install with only _monitor_docker_. Seems to work with my first test  !

FYI here's the logs I can see when I switch off the remote host
```
2025-01-12 12:21:20.419 ERROR (MainThread) [custom_components.monitor_docker.helpers] [DockerMalban] portainer_agent: Container not available anymore (3a) (DockerError(900, "Cannot connect to Docker Engine via http://192.168.1.30:2375 [Cannot connect to host 192.168.1.30:2375 ssl:default [Connect call failed ('192.168.1.30', 2375)]]")). Retry in 60 seconds
2025-01-12 12:21:28.309 ERROR (MainThread) [custom_components.monitor_docker.helpers] [DockerMalban]: run_docker_info (DockerError(900, "Cannot connect to Docker Engine via http://192.168.1.30:2375 [Cannot connect to host 192.168.1.30:2375 ssl:default [Connect call failed ('192.168.1.30', 2375)]]")). Retry in 60 seconds
2025-01-12 12:21:33.105 ERROR (MainThread) [custom_components.monitor_docker.helpers] [DockerMalban]: run_docker_events loop ended
2025-01-12 12:21:33.914 ERROR (MainThread) [custom_components.monitor_docker.helpers] [DockerMalban] Failed Docker connect (DockerError(900, "Cannot connect to Docker Engine via http://192.168.1.30:2375 [Cannot connect to host 192.168.1.30:2375 ssl:default [Connect call failed ('192.168.1.30', 2375)]]")). Retry in 60 seconds

--> entities are unavailable

2025-01-12 12:22:33.922 ERROR (MainThread) [custom_components.monitor_docker.helpers] [DockerMalban] Failed Docker connect (DockerError(900, "Cannot connect to Docker Engine via http://192.168.1.30:2375 [Cannot connect to host 192.168.1.30:2375 ssl:default [Connect call failed ('192.168.1.30', 2375)]]")). Retry in 60 seconds

--> switch on the remote host

2025-01-12 12:23:33.951 ERROR (MainThread) [custom_components.monitor_docker.sensor] [DockerMalban] portainer_agent: Cannot determine network-available?
2025-01-12 12:23:33.952 ERROR (MainThread) [homeassistant] Error doing job: Unclosed client session (None)

--> entities are back
```

However, I also have one unhandled exception
```
2025-01-12 12:31:29.276 ERROR (Recorder) [homeassistant] Error doing job: Unclosed client session (None)
2025-01-12 12:31:29.277 ERROR (Recorder) [homeassistant] Error doing job: Task exception was never retrieved (None)
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/site-packages/aiohttp/client_proto.py", line 92, in connection_lost
    uncompleted = self._parser.feed_eof()
  File "aiohttp/_http_parser.pyx", line 508, in aiohttp._http_parser.HttpParser.feed_eof
aiohttp.http_exceptions.TransferEncodingError: 400, message:
  Not enough data for satisfy transfer length header.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.13/site-packages/aiodocker/events.py", line 62, in run
    async for data in self.json_stream:
        await self.channel.publish(data)
  File "/usr/local/lib/python3.13/site-packages/aiodocker/jsonstream.py", line 25, in __anext__
    data = yield from self._response.content.readline()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/aiohttp/streams.py", line 352, in readline
    return await self.readuntil()
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/aiohttp/streams.py", line 386, in readuntil
    await self._wait("readuntil")
  File "/usr/local/lib/python3.13/site-packages/aiohttp/streams.py", line 347, in _wait
    await waiter
aiohttp.client_exceptions.ClientPayloadError: Response payload is not completed: <TransferEncodingError: 400, message='Not enough data for satisfy transfer length header.'>
```

Also, the 1.20b1 is still tagged 1.20b0 in the manifest 
https://github.com/ualex73/monitor_docker/blob/master/custom_components/monitor_docker/manifest.json#L9

Thanks

**Comment by ualex73** [comment]
> hello, Tried the 1.20b1 with a remote host on a fresh HA install with only _monitor_docker_. Seems to work with my first test !
> 
> FYI here's the logs I can see when I switch off the remote host
> 
> ```
> 2025-01-12 12:21:20.419 ERROR (MainThread) [custom_components.monitor_docker.helpers] [DockerMalban] portainer_agent: Container not available anymore (3a) (DockerError(900, "Cannot connect to Docker Engine via http://192.168.1.30:2375 [Cannot connect to host 192.168.1.30:2375 ssl:default [Connect call failed ('192.168.1.30', 2375)]]")). Retry in 60 seconds
> 2025-01-12 12:21:28.309 ERROR (MainThread) [custom_components.monitor_docker.helpers] [DockerMalban]: run_docker_info (DockerError(900, "Cannot connect to Docker Engine via http://192.168.1.30:2375 [Cannot connect to host 192.168.1.30:2375 ssl:default [Connect call failed ('192.168.1.30', 2375)]]")). Retry in 60 seconds
> 2025-01-12 12:21:33.105 ERROR (MainThread) [custom_components.monitor_docker.helpers] [DockerMalban]: run_docker_events loop ended
> 2025-01-12 12:21:33.914 ERROR (MainThread) [custom_components.monitor_docker.helpers] [DockerMalban] Failed Docker connect (DockerError(900, "Cannot connect to Docker Engine via http://192.168.1.30:2375 [Cannot connect to host 192.168.1.30:2375 ssl:default [Connect call failed ('192.168.1.30', 2375)]]")). Retry in 60 seconds
> 
> --> entities are unavailable
> 
> 2025-01-12 12:22:33.922 ERROR (MainThread) [custom_components.monitor_docker.helpers] [DockerMalban] Failed Docker connect (DockerError(900, "Cannot connect to Docker Engine via http://192.168.1.30:2375 [Cannot connect to host 192.168.1.30:2375 ssl:default [Connect call failed ('192.168.1.30', 2375)]]")). Retry in 60 seconds
> 
> --> switch on the remote host
> 
> 2025-01-12 12:23:33.951 ERROR (MainThread) [custom_components.monitor_docker.sensor] [DockerMalban] portainer_agent: Cannot determine network-available?
> 2025-01-12 12:23:33.952 ERROR (MainThread) [homeassistant] Error doing job: Unclosed client session (None)
> 
> --> entities are back
> ```
> 
> However, I also have one unhandled exception
> 
> ```
> 2025-01-12 12:31:29.276 ERROR (Recorder) [homeassistant] Error doing job: Unclosed client session (None)
> 2025-01-12 12:31:29.277 ERROR (Recorder) [homeassistant] Error doing job: Task exception was never retrieved (None)
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.13/site-packages/aiohttp/client_proto.py", line 92, in connection_lost
>     uncompleted = self._parser.feed_eof()
>   File "aiohttp/_http_parser.pyx", line 508, in aiohttp._http_parser.HttpParser.feed_eof
> aiohttp.http_exceptions.TransferEncodingError: 400, message:
>   Not enough data for satisfy transfer length header.
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.13/site-packages/aiodocker/events.py", line 62, in run
>     async for data in self.json_stream:
>         await self.channel.publish(data)
>   File "/usr/local/lib/python3.13/site-packages/aiodocker/jsonstream.py", line 25, in __anext__
>     data = yield from self._response.content.readline()
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.13/site-packages/aiohttp/streams.py", line 352, in readline
>     return await self.readuntil()
>            ^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.13/site-packages/aiohttp/streams.py", line 386, in readuntil
>     await self._wait("readuntil")
>   File "/usr/local/lib/python3.13/site-packages/aiohttp/streams.py", line 347, in _wait
>     await waiter
> aiohttp.client_exceptions.ClientPayloadError: Response payload is not completed: <TransferEncodingError: 400, message='Not enough data for satisfy transfer length header.'>
> ```
> 
> Also, the 1.20b1 is still tagged 1.20b0 in the manifest https://github.com/ualex73/monitor_docker/blob/master/custom_components/monitor_docker/manifest.json#L9
> 
> I will continue to do more testing
> 
> Thanks

Thanks for testing, and I updated the manifest.json now (we re-release the 1.20b1).

For this unhandled "aiohttp.client_exceptions.ClientPayloadError" exception seem to come from "Recorder", so I need a bit more logging to pinpoint it towards the monitor_docker code?

**Comment by ualex73** [comment]
It should be fixed, so I will close ticket in 7 days if nobody objects :-) The recorder could be a different issue, so we then should raise a new ticket for it.

**Comment by jerome-labidurie** [comment]
Hello,

This is OK for me. I've done tests upon tests with 1.20b2 and everything seems to work fine.

For the Recorder issue, I got the exception only once. But not logs seems to give a clue around it (with DEBUG logs for monitor_docker)

Thanks

**Comment by ualex73** [comment]
I will close this one. I it reoccurs with recording, please open a new ticket with all information.

**Comment by defaultsecurity** [comment]
@ualex73 
Sorry for getting back to you so late. My systems weren't in sync for a while. I have installed version 1.20b3 and you have successfully solved the problem. Thank you very much for your hard work!
